src/event_object_movement.c

```diff
@@ -0,0 +1,8372 @@
+    {
+        objectEvent = &gObjectEvents[objectEventId];
+        sprite = &gSprites[objectEvent->spriteId];
+        objectEvent->fixedPriority = TRUE;
+        sprite->subpriority = subpriority;
+    }
+}
+
+void sub_805F400(u8 localId, u8 mapNum, u8 mapGroup)
+{
+    u8 objectEventId;
+    struct ObjectEvent *objectEvent;
+
+    if (!TryGetObjectEventIdByLocalIdAndMap(localId, mapNum, mapGroup, &objectEventId))
+    {
+        objectEvent = &gObjectEvents[objectEventId];
+        objectEvent->fixedPriority = FALSE;
+        objectEvent->triggerGroundEffectsOnMove = TRUE;
+    }
+}
+
+void sub_808E82C(u8 localId, u8 mapNum, u8 mapGroup, s16 x, s16 y)
+{
+    u8 objectEventId;
+    struct Sprite *sprite;
+
+    if (!TryGetObjectEventIdByLocalIdAndMap(localId, mapNum, mapGroup, &objectEventId))
+    {
+        sprite = &gSprites[gObjectEvents[objectEventId].spriteId];
+        sprite->pos2.x = x;
+        sprite->pos2.y = y;
+    }
+}
+
+void FreeAndReserveObjectSpritePalettes(void)
+{
+    FreeAllSpritePalettes();
+    gReservedSpritePaletteCount = 12;
+}
+
+static void LoadObjectEventPalette(u16 paletteTag)
+{
+    u16 i = FindObjectEventPaletteIndexByTag(paletteTag);
+
+    if (i != OBJ_EVENT_PAL_TAG_NONE) // always true
+    {
+        sub_805F510(&sObjectEventSpritePalettes[i]);
+    }
+}
+
+void Unused_LoadObjectEventPaletteSet(u16 *paletteTags)
+{
+    u8 i;
+
+    for (i = 0; paletteTags[i] != OBJ_EVENT_PAL_TAG_NONE; i++)
+    {
+        LoadObjectEventPalette(paletteTags[i]);
+    }
+}
+
+static u8 sub_805F510(const struct SpritePalette *spritePalette)
+{
+    if (IndexOfSpritePaletteTag(spritePalette->tag) != 0xFF)
+    {
+        return 0xFF;
+    }
+    return LoadSpritePalette(spritePalette);
+}
+
+void PatchObjectPalette(u16 paletteTag, u8 paletteSlot)
+{
+    u8 paletteIndex = FindObjectEventPaletteIndexByTag(paletteTag);
+
+    LoadPalette(sObjectEventSpritePalettes[paletteIndex].data, 16 * paletteSlot + 0x100, 0x20);
+    sub_8083598(paletteSlot);
+}
+
+void PatchObjectPaletteRange(const u16 *paletteTags, u8 minSlot, u8 maxSlot)
+{
+    while (minSlot < maxSlot)
+    {
+        PatchObjectPalette(*paletteTags, minSlot);
+        paletteTags++;
+        minSlot++;
+    }
+}
+
+static u8 FindObjectEventPaletteIndexByTag(u16 tag)
+{
+    u8 i;
+
+    for (i = 0; sObjectEventSpritePalettes[i].tag != OBJ_EVENT_PAL_TAG_NONE; i++)
+    {
+        if (sObjectEventSpritePalettes[i].tag == tag)
+        {
+            return i;
+        }
+    }
+    return 0xFF;
+}
+
+void LoadPlayerObjectReflectionPalette(u16 tag, u8 slot)
+{
+    u8 i;
+
+    PatchObjectPalette(tag, slot);
+    for (i = 0; gPlayerReflectionPaletteSets[i].tag != OBJ_EVENT_PAL_TAG_NONE; i++)
+    {
+        if (gPlayerReflectionPaletteSets[i].tag == tag)
+        {
+            PatchObjectPalette(gPlayerReflectionPaletteSets[i].data[sCurrentReflectionType], gReflectionEffectPaletteMap[slot]);
+            return;
+        }
+    }
+}
+
+void LoadSpecialObject
```

Comment:
event_object_movement.c:2700-2701: When porting from data/event_object_movement.s, replace this memcpy with array initialization.